src/Components/Panels/ExploreLogsVizPanelMenu.tsx

```diff
@@ -0,0 +1,192 @@
+
+  // If we don't have a query runner, then our panel is within a SceneCSSGridItem, we need to get the query runner from there
+  if (!queryRunner) {
+    const sceneGridItem = sceneGraph.getAncestor(sceneRef, SceneCSSGridItem);
+    const queryProvider = sceneGraph.getData(sceneGridItem);
+
+    if (queryProvider instanceof SceneQueryRunner) {
+      queryRunner = queryProvider;
+    } else {
+      logger.error(new Error('query provider not found!'));
+    }
+  }
+  const uninterpolatedExpr: string | undefined = queryRunner.state.queries[0].expr;
+  const expr = sceneGraph.interpolate(sceneRef, uninterpolatedExpr);
+
+  reportAppInteraction(USER_EVENTS_PAGES.all, USER_EVENTS_ACTIONS.all.open_in_explore_menu_clicked);
+
+  onExploreLinkClick(indexScene, expr);
+};
+
+const getInvestigationLink = (addToExplorations: AddToExplorationButton) => {
+  const links = getPluginLinkExtensions({
+    extensionPointId: ExtensionPoints.MetricExploration,
+    context: addToExplorations.state.context,
+  });
+
+  return links.extensions[0];
+};
+
+const onAddToInvestigationClick = (event: React.MouseEvent, addToExplorations: AddToExplorationButton) => {
+  const link = getInvestigationLink(addToExplorations);
+  if (link && link.onClick) {
+    link.onClick(event);
+  }
+};
+
+function subscribeToAddToExploration(exploreLogsVizPanelMenu: ExploreLogsVizPanelMenu) {
+  const addToExplorationButton = exploreLogsVizPanelMenu.state.addToExplorations;
+  if (addToExplorationButton) {
+    const link = getInvestigationLink(addToExplorationButton);
+    const disabledLinks = addToExplorationButton.state.disabledLinks;
+
+    const existingMenuItems = exploreLogsVizPanelMenu.state.body?.state.items ?? [];
+
+    const existingAddToExplorationLink = existingMenuItems.find((item) => item.text === ADD_TO_INVESTIGATION_MENU_TEXT);
+
+    if (!(link.category === 'disabled' || disabledLinks.includes(link.id)) && !existingAddToExplorationLink) {
+      exploreLogsVizPanelMenu.state.body?.addItem({
+        text: ADD_TO_INVESTIGATION_MENU_DIVIDER_TEXT,
+        shortcut: '',
+        type: 'divider',
+      });
+      exploreLogsVizPanelMenu.state.body?.addItem({
+        text: ADD_TO_INVESTIGATION_MENU_TEXT,
+        iconClassName: 'plus-square',
+        onClick: (e) => onAddToInvestigationClick(e, addToExplorationButton),
+      });
+    } else {
+      if (existingAddToExplorationLink) {
+        exploreLogsVizPanelMenu.state.body?.setItems(
+          existingMenuItems.filter(
+            (item) =>
+              item.text !== ADD_TO_INVESTIGATION_MENU_TEXT && item.text !== ADD_TO_INVESTIGATION_MENU_DIVIDER_TEXT
+          )
+        );
+      }
+    }
+  }
+}
+
+export const getPanelWrapperStyles = (theme: GrafanaTheme2) => {
+  return {
+    panelWrapper: css({
+      width: '100%',
+      height: '100%',
+      label: 'panel-wrapper',
+      display: 'flex',
+
+      // Need more specificity to override core style
+      'button.show-on-hover': {
```

Comment:
Might want to add e2e test asserting that the menus are visible without hovering. Also might be a good idea to open a core issue/PR to allow consumers of VizPanelMenu to define if the menu is always visible, or only visible on hover

Reply:
I changed my mind, hopefully https://github.com/grafana/grafana/issues/96865 makes progress soon and this functionality will be tested in core, where it should be